doc/rtd/topics/code_review.rst

```diff
@@ -0,0 +1,170 @@
+*******************
+Code Review Process
+*******************
+
+In order to manage incoming pull requests effectively, and to ensure
+that community members (like you!) submitting them understand what to
+expect, cloud-init has a documented code review process.
+
+The language used is intentionally formal, in a bid to avoid ambiguity,
+but the *experience* of working through the code review process should
+not be, so please don't be put off!  If you have any questions about
+this process, please do ask them in ``#cloud-init`` on the Freenode IRC
+network.
+
+Goals
+=====
+
+This process has the following goals:
+
+* Ensure that cloud-init community members are receiving code reviews
+  in a timely fashion
+* Minimize the time the cloud-init core developers have to spend on the
+  parts of the code review process which aren't actual code review
+
+Roles
+=====
+
+For ease of understanding, we will refer to two different roles
+throughout the process:
+
+Proposer
+   The developer proposing a pull request (hereafter known as a PR).
+   This could be a core developer or community member.  (This is likely
+   the role you will be taking in this process.)
+
+the Committers
+   The group of cloud-init core developers.
+
+   In this process doc, cloud-init core developers are treated as a
+   bloc of interchangeable people.  This is a simplification, as in
+   reality we will need to manage transfer of reviews, dismissal of
+   stale reviews, parallel reviews from multiple committers, etc.  If
+   we find that handling these on an ad-hoc basis is causing problems,
+   then we should revisit and expand this process to include them.
+
+   (If the **Proposer** is a core developer, then they are not
+   considered part of this group for the purposes of this process.)
+
```

Comment:
The way this is structure unfortunately discourages others from reviewing code, IMHO. The core team should have keen interest in training and enabling additional core team members. Doing code reviews is one of the primary training tools for such growth. If code reviews by others are more or less actively discouraged by way of the implications as written here it is going to make things more difficult.

Over all I would suggest that we switch the use of "Committer" to "Reviewer" in the context of reviews. This then allows for clearer separation of the tasks of review and "merge/acceptance" in that the group "Reviewer" and "Committer" get separate definitions. Effectively we would have 3 role definitions in this section.

At some point during the process a "Committer" takes on the role of "Reviewer" but really anyone, spare the "Proposer" should be able to assume the role of "Reviewer" and that voice should be heard. Obviously a review by a core team member is required for the merge. That can be stated in a way that does not discourage others from being "Reviewers".

Reply:
That's a great point, thank you!